catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceSource.java

```diff
@@ -0,0 +1,386 @@
+                    || response.getStatus() == HttpStatus.SC_PAYMENT_REQUIRED)) {
+                isAvailable = true;
+                lastAvailableDate = new Date();
+            }
+        } else {
+            isAvailable = lastAvailable;
+        }
+
+        if (lastAvailable != isAvailable) {
+            for (SourceMonitor monitor : this.sourceMonitors) {
+                if (isAvailable) {
+                    monitor.setAvailable();
+                } else {
+                    monitor.setUnavailable();
+                }
+            }
+        }
+
+        lastAvailable = isAvailable;
+        return isAvailable;
+    }
+
+    @Override
+    public boolean isAvailable(SourceMonitor callback) {
+        sourceMonitors.add(callback);
+        return isAvailable();
+    }
+
+    @Override
+    public SourceResponse query(QueryRequest request) throws UnsupportedQueryException {
+        SearchResource confluence = factory.getClient();
+        Query query = request.getQuery();
+        ConfluenceFilterDelegate confluenceDelegate = new ConfluenceFilterDelegate();
+
+        String cql = filterAdapter.adapt(query, confluenceDelegate);
+        if (!confluenceDelegate.isConfluenceQuery() || (StringUtils.isEmpty(cql) && (
+                confluenceSpaces.isEmpty() || !confluenceDelegate.isWildCardQuery()))) {
+            return new SourceResponseImpl(request, Collections.emptyList());
+        }
+        cql = getSortedQuery(query.getSortBy(), getSpaceQuery(cql));
+        LOGGER.debug(cql);
+
+        String finalExpandedSections = expandedSections;
+        if (includePageContent) {
+            finalExpandedSections += ",body.view";
+        }
+
+        String cqlContext = null;
+        String excerpt = null;
+        Response confluenceResponse = confluence.search(cql,
+                cqlContext,
+                excerpt,
+                finalExpandedSections,
+                query.getStartIndex() - 1,
+                query.getPageSize(),
+                includeArchivedSpaces);
+
+        InputStream stream = null;
+        Object entityObj = confluenceResponse.getEntity();
+        if (entityObj != null) {
+            stream = (InputStream) entityObj;
+        }
+        if (Response.Status.OK.getStatusCode() != confluenceResponse.getStatus()) {
+            String error = "";
+            try {
+                if (stream != null) {
+                    error = IOUtils.toString(stream);
+                }
+            } catch (IOException ioe) {
+                LOGGER.debug("Could not convert error message to a string for output.", ioe);
+            }
+            throw new UnsupportedQueryException(String.format(
+                    "Received error code from remote source (status %s ): %s",
+                    confluenceResponse.getStatus(),
+                    error));
+        }
+
+        try {
+
+            List<Result> results = transformer.transformConfluenceResponse(stream)
```

Comment:
Because you are relying on a specific implementation detail that only the ConfluenceInputTransformer provides, no one would be able to override the implementation.

I think it would be more appropriate to move the logic that does this transformation into a `MessageBodyReader`. It can handle the transformation of the "wrapper" around the results and still delegate the transformation to an InputTransformer that abides by the interface, allowing a new one to be plugged in if needed.